set_bucket_tags/app.py

```diff
@@ -55,31 +59,94 @@ def get_principal_id(tags):
 
 def get_synapse_user_profile(synapse_id):
   '''Get synapse user profile data'''
-  syn = synapseclient.Synapse()
+  syn = get_synapse_client()
   user_profile = syn.getUserProfile(synapse_id)
   log.debug(f'Synapse user profile: {user_profile}')
-
   return user_profile
 
 
-def get_synapse_tags(user_profile):
-  '''Derive synapse tags from synapse user profile data'''
-
+def get_ssm_parameter(name):
+  '''Get an parameter from the SSM parameter store'''
+  client = boto3.client('ssm')
+  parameter = client.get_parameter(Name=name)
+  log.debug(f'Synapse ssm parameter: {parameter}')
+  return parameter
+
+
+def get_synapse_team_ids():
+  '''Get synapse team IDs'''
+  TeamToRoleArnMap = os.getenv('TEAM_TO_ROLE_ARN_MAP_PARAM_NAME')
+  ssm_param = get_ssm_parameter(TeamToRoleArnMap)
+  team_to_role_arn_map = json.loads(ssm_param["Parameter"]["Value"])
+  log.debug(f'/service-catalog/TeamToRoleArnMap value: {team_to_role_arn_map}')
+  team_ids = []
+  for item in team_to_role_arn_map:
+    team_ids.append(item["teamId"])
+
+  log.debug(f'Synapse team IDs: {team_ids}')
+  return team_ids
+
+
+def get_synapse_user_team_id(synapse_id, team_ids):
+  '''Get the Synapse team that the user is in
+  :param synapse_id: synapse user id
+  :param team_ids: the synapse team ids
+  :returns: the synapse team id that the user is in, None if user is not
```

Comment:
the id of the first synapse team in the given list that the user is in, None is user is not...

Reply:
fixed